crates/sim2h/src/lib.rs

```diff
@@ -998,16 +1181,26 @@ impl Sim2h {
 
         let payload: Opaque = msg.clone().into();
 
-        match self.open_connections.get_mut(&uri) {
-            None => {
-                error!("FAILED TO SEND, NO ROUTE: {}", uri);
-                return;
-            }
-            Some((_con, outgoing_send)) => {
+        let get_result = open_connections.get_mut(&uri);
```

Comment:
I don't think this is going to do what you want... [get_mut](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.get_mut) returns a None if somebody else currently has a lock, in which case it'll print out the NO ROUTE error message below and exit, which isn't right.

Reply:
n/m it calling the get_mut on the map, not the arc : )